mlflow/gateway/config.py

```diff
@@ -298,26 +295,31 @@ def validate_provider(cls, value):
         raise MlflowException.invalid_parameter_value(f"The provider '{value}' is not supported.")
 
     @classmethod
-    def _validate_config(cls, info, values):
+    def _validate_config(cls, val, context):
         from mlflow.gateway.provider_registry import provider_registry
 
-        if provider := values.get("provider"):
-            config_type = provider_registry.get(provider).CONFIG_TYPE
-            return config_type(**info)
+        # For Pydantic v2: 'context' is a ValidationInfo object with a 'data' attribute.
+        # For Pydantic v1: 'context' is dict-like 'values'.
+        if IS_PYDANTIC_V2_OR_NEWER:
+            provider = context.data.get("provider")
+        else:
+            provider = context.get("provider") if context else None
```

Comment:
I would also move this into the pydantic_utils, so that the day we want to finally remove pydantic v1, we would just need to delete the pydantic_utils file